### ISSUE TITLE
CORS policy: Response to preflight request doesn't pass access control check

### ISSUE THREAD

**Issue by samuelgrieve1** [grokability/snipe-it#13378]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

This works fine when I use Postman, but I'm trying to call the api from a local web server and am getting this response:

Access to fetch at 'https://lfsf.snipe-it.io/api/v1/hardware' from origin 'http://127.0.0.1:5500' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Does anyone know what the issue may be?

This is my JS file:

```
const options = {
  method: 'GET',
  headers: {
    accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: 'Bearer KEY_REDACTED'
  }
};

fetch('https://MY_DOMAIN.snipe-it.io/api/v1/hardware', options)
  .then(response => response.json())
  .then(response => console.log(response))
  .catch(err => console.error(err))
```

### Reproduction steps

1. Load webpage
2. Go to inspector
3. See error
...

### Expected behavior

Expect a response without errors.

### Snipe-IT Version

Version v6.1.2 - build 10938

### Operating System

MacOS

### Web Server

Node

### PHP Version

Unknown

**Comment by snipe** [comment]
IIRC, you should be able to add `CORS_ALLOWED_ORIGINS="127.0.0.1:5500"` to your .env file to allow localhost. CORS can be a bit fiddly though, so you may have to experiment a little with adding the protocol and/or removing the port number.

https://github.com/snipe/snipe-it/blob/642a09f139f3f7ceda38bc1d8a88b3c9f3cfe157/config/cors.php#L19-L20

**Comment by snipe** [comment]
Hi there - We haven't heard back in a bit, so I'm going to close this ticket for now, but will re-open it if you're still having issues.

![hackerman-hacking](https://user-images.githubusercontent.com/197404/31936003-27a7f6d8-b865-11e7-8f5e-eb03062925f4.gif)

**Comment by aQkiRa** [comment]
Hi,

I'm currently running the Snipe-IT app and API locally using XAMPP, and everything works perfectly. I then deployed Snipe-IT to a cPanel server, and the application itself runs fine there as well. However, I ran into a problem when trying to integrate Snipe-IT's API with my web app — specifically with CORS (Cross-Origin Resource Sharing).

Authentication:
I created an API integration where users from my app are synced with users in Snipe-IT. I’m using a super admin token to retrieve user data. This works perfectly in my local environment, so I proceeded to deploy it to production (on cPanel).

My Web App:
My frontend is built with JavaScript. Here's a simplified example of the code I use:
const url = `${api.protocolHostname}/${endpoint}`;
const response = await fetch(url, {
    method: 'POST',
    headers: this.getAuthHeaders(),
    body: JSON.stringify(data),
    credentials: 'include' // Important for cookies
});

Issue:
In production, I’m getting CORS errors when trying to access the Snipe-IT API from my frontend. I’ve tried several solutions including setting CORS_ALLOWED_ORIGINS="sdla.mydomain.com" in the .env file as suggested [here](https://github.com/snipe/snipe-it/blob/642a09f139f3f7ceda38bc1d8a88b3c9f3cfe157/config/cors.php#L19-L20). But I’m still having the same issue.



<img width="1200" height="876" alt="Image" src="https://github.com/user-attachments/assets/42b35033-87be-450e-814a-21e849506c19" />
